pontoon/base/views.py

```diff
@@ -76,34 +81,81 @@ def home(request):
     return translate(request, locale, project.slug)
 
 
-def locale(request, locale, template='locale.html'):
-    """Locale view."""
-    log.debug("Locale view.")
+class TemplateModelView(TemplateView):
```

Comment:
This is essentially a simplified reimplementation of the built-in [DetailView](https://ccbv.co.uk/projects/Django/1.8/django.views.generic.detail/DetailView/). Was that intentional?

In any case, I think catching the exception this far up in the call stack isn't wise, as an `ObjectDoesNotExist` error is not always because the main object we're trying to fetch couldn't be found. It could be another type of error, maybe fetching a related object, and this would swallow that error up without us knowing. That's why `DetailView` only catchs the `ObjectDoesNotExist` error within `get_object` and not everywhere.